marlowe-playground-client/src/Marlowe/Holes.purs

```diff
@@ -189,6 +189,18 @@ getMarloweConstructors PartyType =
 allMarloweConstructors :: Map String (Array Argument)
 allMarloweConstructors = foldl (\acc mt -> getMarloweConstructors mt <> acc) mempty allMarloweTypes
 
+allMarloweConstructorNames :: Map String MarloweType
+allMarloweConstructorNames = foldl (\acc t -> f t <> acc) mempty allMarloweTypes
+  where
+  f :: MarloweType -> Map String MarloweType
+  f marloweType' =
+    let
+      constructors = Map.keys $ getMarloweConstructors marloweType'
+
+      (kvs :: Array (Tuple String MarloweType)) = Set.toUnfoldable $ Set.map (\constructor -> Tuple constructor marloweType') constructors
```

Comment:
Picky, but this is already such a long line that:

```purescript
kvs :: Array (Tuple String MarloweType)
kvs = Set.toUnfoldable $ Set.map (\constructor -> Tuple constructor marloweType') constructors
```

...would be much better.